src/main/java/com/fasterxml/jackson/databind/introspect/AnnotatedField.java

```diff
@@ -128,6 +128,14 @@ public Object getValue(Object pojo) throws IllegalArgumentException
      */
     public boolean isTransient() { return Modifier.isTransient(getModifiers()); }
 
+    /**
+     * @since 2.18
+     */
+    public boolean isImmutable() {
+        // Records' fields can't mutated via reflection (JDK-8247517)
+        return _typeContext.resolveType(getDeclaringClass()).isRecordType();
```

Comment:
This seems like a wrong way to achieve this. I'll add a suggestion above for alternative in caller.